Firebase.cpp

```diff
@@ -74,21 +79,32 @@ String Firebase::makeURL(const String& path) {
   return url;
 }
 
-String Firebase::sendRequest(const char* method, const String& path, const String& value) {
-  _error.reset();
+int Firebase::sendRequest(const char* method, const String& path, const String& value) {
   String url = makeURL(path);
   _http.begin(_host.c_str(), firebasePort, url.c_str(), true, firebaseFingerprint);
-  int statusCode = _http.sendRequest(method, (uint8_t*)value.c_str(), value.length());
-  if (statusCode < 0) {
-    _error.set(statusCode,
-	       String(method) + " " + url + ": "
-	       + HTTPClient::errorToString(statusCode));
+  int statusCode =  _http.sendRequest(method, (uint8_t*)value.c_str(), value.length());
+  setError(method, url, statusCode);
+  return statusCode;
+}
+
+String Firebase::sendRequestGetBody(const char* method, const String& path, const String& value) {
+  sendRequest(method, path, value);
+  if (_error.code() != 0) {
     return "";
   }
   // no _http.end() because of connection reuse.
   return _http.getString();
```

Comment:
Sounds good. We will continue discussion of higher level refactoring in the other PR.